src/Cli/dotnet/commands/dotnet-run/VirtualProjectBuildingCommand.cs

```diff
@@ -117,6 +131,31 @@ public int Execute(string[] binaryLoggerArgs, ILogger consoleLogger)
         }
     }
 
+    /// <summary>
+    /// Needs to be called before the first call to <see cref="CreateProjectInstance(ProjectCollection)"/>.
+    /// </summary>
+    public VirtualProjectBuildingCommand PrepareProjectInstance()
+    {
+        Debug.Assert(_directives.IsDefault && _targetFilePath is null, $"{nameof(PrepareProjectInstance)} should not be called multiple times.");
+
+        var sourceFile = CreateSourceFile(EntryPointFileFullPath);
+        _directives = FindDirectives(sourceFile);
+
+        // If there were any `#:` directives, remove them from the file.
+        // (This is temporary until Roslyn is updated to ignore them.)
+        _targetFilePath = EntryPointFileFullPath;
+        if (_directives.Length != 0)
+        {
+            var targetDirectory = Path.Join(Path.GetDirectoryName(_targetFilePath), "obj");
+            Directory.CreateDirectory(targetDirectory);
```

Comment:
Is the idea to make a .targets holding the directives information? That seems an odd choice to me compared with just putting them in the csproj

Reply:
No, this has nothing to do with `.targets`. I'm assuming you are referring to `targetDirectory`. Given `dotnet run ./app/file.cs`, targetDirectory will be `./app/obj/` where we will place `./app/obj/file.cs` that has the directives stripped out so we can pass it to Roslyn. This is just temporary, we will skip all this when Roslyn recognizes the directives (i.e., doesn't error on them).